### PR TITLE
Add TestGenerateMojo to the akka-grpc-maven-plugin #1099

### PR DESCRIPTION
**Pull request by samantmaharaj**
References #1099

### CODE REVIEW

**Comment by lightbend-cla-validator** [comment]
Hi @samantmaharaj,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Lightbend Contributors License Agreement:

[https://www.lightbend.com/contribute/cla](https://www.lightbend.com/contribute/cla)

**Comment by raboof** [comment]
OK TO TEST

File: maven-plugin/src/main/maven/plugin.xml [hunk 1]
```diff
@@ -79,6 +86,90 @@
...
+      </parameters>
+      <configuration>
+        <project implementation="" default-value="${project}"/>
+        <language implementation="enum" default-value="JAVA">${akka-grpc.language}</language>
+        <generateClient implementation="boolean" default-value="true">${akka-grpc.generate-client}</generateClient>
+        <generateServer implementation="boolean" default-value="true">${akka-grpc.generate-server}</generateServer>
+        <generatorSettings implementation="java.util.List" default-value=""/>
+        <extraGenerators implementation="java.util.List" default-value=""/>
+        <protoPaths default-value="src/test/proto,src/test/protobuf">${akka-grpc.protoPaths}</protoPaths>
+        <outputDirectory default-value="target/generated-test-sources">${akka-grpc.outputDirectory}</outputDirectory>
```
Comment (raboof) [inline maven-plugin/src/main/maven/plugin.xml:172]: Hmm, that's quite some duplication.

Is this the way other code-generating plugins do it as well?

File: maven-plugin/src/main/scala/akka/grpc/maven/TestGenerateMojo.scala [hunk 2]
```diff
@@ -0,0 +1,14 @@
+/*
+ * Copyright (C) 2018-2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.grpc.maven
+
+import javax.inject.Inject
+import org.sonatype.plexus.build.incremental.BuildContext
+
+class TestGenerateMojo @Inject() (buildContext: BuildContext) extends AbstractGenerateMojo(buildContext) {
```
Comment (raboof) [inline maven-plugin/src/main/scala/akka/grpc/maven/TestGenerateMojo.scala:10]: :+1:

File: maven-plugin/src/main/maven/plugin.xml (hunk 1 above)
Comment (samantmaharaj) [reply maven-plugin/src/main/maven/plugin.xml:172]: If you develop using maven, the `plugin.xml` is generated based on annotations in the code so it's not a problem that normally comes up as you don't need to maintain it.

I could factor the common XML out and use a templating tool but I'm not that familiar with SBT so I'd appreciate some help on how to do it.

Comment (raboof) [reply maven-plugin/src/main/maven/plugin.xml:172]: Aah makes sense. Adding templating would be possible for example with sbt-twirl, but to be honest I'm not sure the added complexity is worth it. Let's live with the duplication for now ;)

**Pull request merged**